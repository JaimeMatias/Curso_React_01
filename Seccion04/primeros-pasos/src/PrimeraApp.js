import PropTypes from 'prop-types';


const PrimeraApp = ({ saludo, edad}) => {
    //Todas las funciones  pueden devolver 1 unico elemento, para devolver varias etiquetas desde una función
    //Para poder devolver varias etiquetas, se utiliza el elemento  de abrir y cerrar llave

    const saludo1 = {
        nombre: 'Matias',
        Apellido: 'Jaime',
    }; // Para poder mostrar variables en el HTML, hay que usar los corchetes {}
    // Para poder mostrar un objeto, hay que primero pasarlo a Json

    return <>
        <pre>{JSON.stringify(saludo1.nombre)}</pre>
        <h1>Mi primera Aplicación</h1>
        <h1>{edad}</h1>
    </>

};

//Sirve para definir las propiedades del componente eL Tipo del valor y si es requerido o no
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    edad: PropTypes.number,
}

//Sirve para definir los valores por defecto de las distintas propiedades del componente
PrimeraApp.defaultProps = {
    saludo: 'Hola', 
    edad: 15,
}
export default PrimeraApp;