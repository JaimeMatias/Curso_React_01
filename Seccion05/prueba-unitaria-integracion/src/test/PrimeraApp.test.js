import { render } from "@testing-library/react"
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimerApp />',()=>{

    //Para que funcione es necesario sacar el comentario en SetupTests
    test('debe mostrar el mensaje "Matias" ', () => {
      const saludo='"Matias"';
      const {getByText}=render(<PrimeraApp />)

      //wqexpect(getByText(saludo)).toBeInTheDocument()

    })

    test('Debe de mostrar <PrimerApp /> correctamente', () => {
      const wrapper =shallow( <PrimeraApp />);

      expect(wrapper).toMatchSnapshot();
    })
    
    
})