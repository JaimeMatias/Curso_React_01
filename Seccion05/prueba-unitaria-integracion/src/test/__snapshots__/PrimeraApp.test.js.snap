// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimerApp /> Debe de mostrar <PrimerApp /> correctamente 1`] = `
<Fragment>
  <h1>
    "Matias"
  </h1>
  <p>
     
    15
     
  </p>
  <h1>
    Mi primera Aplicaci√≥n
  </h1>
</Fragment>
`;
